server/src/lib/server.rs

```diff
@@ -60,16 +60,19 @@ enum SerialTaskError {
 struct SerialTask {
     /// Handle to attached serial session
     task: JoinHandle<()>,
-    /// Oneshot channel used to detach an attached serial session
-    detach_ch: oneshot::Sender<()>,
+    /// Oneshot channel used to signal the task to terminate gracefully
+    close_ch: Option<oneshot::Sender<()>>,
```

Comment:
Why does this need to be an `Option`?

Reply:
`oneshot::Sender::send` takes ownership of `self`, `Drop::drop` doesn't, and it was the cleanest thing that came to mind to deal with that situation